File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/scenario/Scenario.java
package com.gmail.borlandlp.minigamesdtools.arena.scenario;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent;

public interface Scenario extends ArenaPhaseComponent {
    boolean isDone();

    boolean roundMustBeEnded();
    boolean gameMustBeEnded();
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/scenario/Scenario.kt
package com.gmail.borlandlp.minigamesdtools.arena.scenario

import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent

interface Scenario : ArenaPhaseComponent {
    val isDone: Boolean
    fun roundMustBeEnded(): Boolean
    fun gameMustBeEnded(): Boolean
}